pyro/infer/autoguide/guides.py

```diff
@@ -394,12 +439,19 @@ def _setup_prototype(self, *args, **kwargs):
         # Initialize guide params
         for name, site in self.prototype_trace.iter_stochastic_nodes():
             # Collect unconstrained event_dims, which may differ from constrained event_dims.
-            init_loc = biject_to(site["fn"].support).inv(site["value"].detach())
+            init_loc = biject_to(site["fn"].support).inv(site["value"].detach()).detach()
```

Comment:
why double `detach()`? wrap in`no_grad()` instead?

Reply:
Even if we wrapped in `no_grad()` we'd need to detach, since `biject_to()` might be simply `lambda x: x`.